src/CMakeLists.txt

```diff
@@ -635,6 +635,9 @@ target_include_directories(MIOpen PUBLIC
 
 target_include_directories(MIOpen SYSTEM PUBLIC $<BUILD_INTERFACE:${HALF_INCLUDE_DIR}>)
 target_include_directories(MIOpen SYSTEM PRIVATE ${BZIP2_INCLUDE_DIR})
+if(MIOPEN_USE_MLIR)
+target_include_directories(MIOpen SYSTEM PRIVATE ${MLIRMIOpen_INCLUDE_DIR})
```

Comment:
The includes should come from the cmake targets, not from cmake variables.

Reply:
The variable `MLIRMIOpen_INCLUDE_DIR` is set in `MiirConfig.cmake`, which is auto-generated when building the MLIR project ([see code here](https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/blob/484f2897d1a4c7b54802cc83ab2d04bec41b8ed8/mlir/tools/mlir-miopen-lib/cmake/MiirConfig.cmake.in#L4)) and loaded first by `find_package(Miir)`. Is this what you are suggesting @pfultz2 ?

Reply:
We shouldn't use the variables such as `MLIRMIOpen_INCLUDE_DIR` as they dont contain all the usage requirements. We should just use the cmake targets, so `target_link_libraries(MIOpen PRIVATE Miir::MLIRMIOpen)` should enough. If thats not the case, then its a bug in mlir which needs to be fixed, and we shouldn't need a workaround.

Reply:
The new MLIR installs the static library as an exported target. Therefore, it needs to tell MIOpen targets the location of the header file `Miir.h`.

Reply:
I think @pfultz2 wants to say that we shouldn't do this manually. If the package was assembled correctly, after finding it cmake can automatically transfer "public dependencies of MLIRMIOpen" to MIOpen if MLIR was part of 'target_link_libraries'.

Reply:
@pfultz2 @shurale-nkn Thank you for the explanation. I got the include_dir information embedded into the export target as its property so that the `target_link_libraries()` is the only needed by MIOpen to use libMLIRMIOpen.